test/functional/workgroup/tests/test-workgroup-Unordered.hpp

```diff
@@ -0,0 +1,339 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing tests for RAJA workgroup unordered runs.
+///
+
+#ifndef __TEST_WORKGROUP_UNORDERED__
+#define __TEST_WORKGROUP_UNORDERED__
+
+#include "RAJA_test-workgroup.hpp"
+#include "RAJA_test-forall-data.hpp"
+
+#include <random>
+
+
+template <typename T>
+class WorkGroupBasicUnorderedSingleFunctionalTest : public ::testing::Test
+{
+};
+
+TYPED_TEST_SUITE_P(WorkGroupBasicUnorderedSingleFunctionalTest);
+
+template <typename T>
+class WorkGroupBasicUnorderedMultipleFunctionalTest : public ::testing::Test
+{
+};
+
+TYPED_TEST_SUITE_P(WorkGroupBasicUnorderedMultipleFunctionalTest);
+
+
+template <typename ExecPolicy,
+          typename OrderPolicy,
+          typename StoragePolicy,
+          typename IndexType,
+          typename Allocator,
+          typename WORKING_RES
+          >
+void testWorkGroupUnorderedSingle(IndexType begin, IndexType end)
+{
+  ASSERT_GE(begin, (IndexType)0);
+  ASSERT_GE(end, begin);
+  IndexType N = end + begin;
+
+  camp::resources::Resource working_res{WORKING_RES()};
+
+  IndexType* working_array;
+  IndexType* check_array;
+  IndexType* test_array;
+
+  allocateForallTestData<IndexType>(N,
+                                    working_res,
+                                    &working_array,
+                                    &check_array,
+                                    &test_array);
+
+  for (IndexType i = IndexType(0); i < N; i++) {
+    test_array[i] = IndexType(0);
+  }
+
+  working_res.memcpy(working_array, test_array, sizeof(IndexType) * N);
+
+  for (IndexType i = begin; i < end; ++i) {
+    test_array[ i ] = IndexType(i) + IndexType(5);
+  }
+
+  RAJA::WorkPool<
+                  RAJA::WorkGroupPolicy<ExecPolicy, OrderPolicy, StoragePolicy>,
+                  IndexType,
+                  RAJA::xargs<>,
+                  Allocator
+                >
+      pool(Allocator{});
+
+  pool.enqueue(RAJA::TypedRangeSegment<IndexType>{ begin, end },
+      [=] RAJA_HOST_DEVICE (IndexType i) {
+    working_array[i] += i + IndexType(5);
+  });
+
+  auto group = pool.instantiate();
+
+  auto site = group.run();
```

Comment:
I guess there's no value in checking the returned "site" because it is a placeholder for each run?

Reply:
ya, the WorkSite objects don't have any accessor methods for their state and none of them really have any state currently anyway.

Reply:
Can there be a situation where pool.instantiate() returns an invalid group?

Reply:
There shouldn't be, you can get an empty group however.